### PR TITLE
Add PaymentRequestEvent.changePaymentMethod(methodName, details) test.

### PR DESCRIPTION
**Pull request by rsolomakhin**
This pull request adds a payment handler test for PaymentRequestEvent.changePaymentMethod(methodName, details) method.

### CODE REVIEW

**Comment by rsolomakhin** [comment]
@danyao, @marcoscaceres : Please take a look!

**Review by danyao — APPROVED** [review]
LGTM!

File: payment-handler/app-change-payment-method.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+self.addEventListener('canmakepayment', event => {
+  event.respondWith(true);
+});
+
+self.addEventListener('paymentrequest', event => {
```
Comment (marcoscaceres) [inline payment-handler/app-change-payment-method.js]: Alternative, using async/await... might be little easier to read/debug if things go bad (hopefully didn't screw anything up refactoring it 🤞):
```JS
self.addEventListener("canmakepayment", event => {
  event.respondWith(true);
});

async function responder(event) {
  const methodName = event.methodData[0].supportedMethods;
  if (!event.changePaymentMethod) {
    return {
      methodName,
      changePaymentMethodReturned:
        "The changePaymentMethod() method is not implemented.",
    };
  }
  let changePaymentMethodReturned;
  try {
    const response = await event.changePaymentMethod(methodName, {
      country: "US",
    });
    changePaymentMethodReturned = response;
  } catch (err) {
    changePaymentMethodReturned = error.message;
  }
  return { methodName, changePaymentMethodReturned };
}

self.addEventListener("paymentrequest", event => {
  event.respondWith(responder(event));
});
```

File: payment-handler/change-payment-method.https.html [hunk 2]
```diff
@@ -0,0 +1,122 @@
+<!doctype html>
```
Comment (marcoscaceres) [inline payment-handler/change-payment-method.https.html]: Just a minor thing, I'd suggest using https://prettier.io to tidy up the code.

**Review by marcoscaceres — APPROVED** [review]
couple of suggestions, but the tests themselves look good.

File: payment-handler/change-payment-method.https.html (hunk 2 above)
Comment (rsolomakhin) [reply payment-handler/change-payment-method.https.html]: Used this command:

```bash
yarn prettier \
  --single-quote \
  --trailing-comma es5 \
  --no-bracket-spacing \
  --jsx-bracket-same-line \
  --arrow-parens always \
  --write <file-name> 
```

File: payment-handler/app-change-payment-method.js (hunk 1 above)
Comment (rsolomakhin) [reply payment-handler/app-change-payment-method.js]: Thank you! Applied your suggestion, @marcoscaceres . Could you review the pull request again, please?

**Review by marcoscaceres — APPROVED** [review]

**Comment by marcoscaceres** [comment]
Not sure about the task cluster failures. Might just need a rebase?

**Comment by rsolomakhin** [comment]
@marcoscaceres wrote:

> Not sure about the task cluster failures. Might just need a rebase?

The failure is due to https://github.com/web-platform-tests/wpt/issues/16448.

**Pull request merged**